generators/java/sdk/src/main/java/com/fern/java/client/generators/OAuthTokenSupplierGenerator.java

```diff
@@ -164,18 +172,31 @@ public GeneratedJavaFile generateFile() {
                 .addStatement(
                         "this.$L = authResponse.get$L()", ACCESS_TOKEN_FIELD_NAME, accessTokenResponsePropertyName);
         if (refreshRequired) {
-            String tokenPropertyName = expiryResponseProperty
-                    .get()
+            ResponseProperty expiresInProperty = expiryResponseProperty.get();
+            String tokenPropertyName = expiresInProperty
                     .getProperty()
                     .getName()
                     .getName()
                     .getPascalCase()
                     .getUnsafeName();
-            getMethodSpecBuilder.addStatement(
-                    "this.$L = $L(authResponse.get$L())",
-                    EXPIRES_AT_FIELD_NAME,
-                    GET_EXPIRES_AT_METHOD_NAME,
-                    tokenPropertyName);
+            TypeReference expiresInType = expiresInProperty.getProperty().getValueType();
+            boolean isOptional = isOptionalType(expiresInType);
+            if (isOptional) {
+                // Handle optional expires_in with default fallback
+                // In Java, optional fields return Optional<T>, so use .orElse()
+                getMethodSpecBuilder.addStatement(
+                        "this.$L = $L(authResponse.get$L().orElse((int) $L))",
```

Comment:
what does this code look like? Is it `.orElse((int) "3600")`?

I think it should just be `.orElse($N)` which results in `.orElse(3600)`